chicken-sloths/bangazon-api-sprint1
# all methods on /computers endpoints

# Description
Establishes the following endpoints:
- `GET /computers`: get all computres
- `GET /computers/:id`: get one computer by id
- `POST /computers`: post new computer
- `PUT /computers/:id`: update one computer by id
- `DELETE /computers/:id`: delete one computer by id

## Related Ticket(s)
fixes #56, extends #88, fixes #8 

## Steps to Test Solution
1. `git checkout kb-computer-post`
1. `npm install`
1. comment out non-computer routes
1. `npm run db:generate`
1. `npm start`
1. http://localhost:8080/api/v1/computers should give you 25 computers
1. http://localhost:8080/api/v1/computers/20 should give you
  ```js
  {
  computer_id: 20,
  mac_address: "42:c6:54:a1:77:3d",
  purchase_date: "2016-04-06T01:32:38.272Z",
  decommission_date: "2017-11-10T14:25:59.680Z"
  }
  ```
1. open Postman or whatever and `POST localhost:8080/api/v1/computers`
  ```json
  { "mac_address": "hello",
  "purchase_date": "my name is",
  "decommission_date": "inigo montoya" }
  ```
1. http://localhost:8080/api/v1/computers/25 should now show the above JSON
1. open Postman or whatever and `PUT http://localhost:8080/api/v1/computers/20`
  ```json
  { "mac_address": "hello",
  "purchase_date": "my name is",
  "decommission_date": "inigo montoya" }
  ```
1. http://localhost:8080/api/v1/computers/20 should now also show that JSON
1. open Postman or whatever and `DELETE localhost:8080/api/v1/computers/25`
1. http://localhost:8080/api/v1/computers/25 should now give you a "Computer not found." error
